### ISSUE TITLE
What happen when change optimizer.lr locally without using kvstore.set_optimizer to update?

### ISSUE THREAD

**Issue by huyangc** [apache/incubator-mxnet#8907]
I am running on a single machine with multi gpus. And the kvstore is set to be device.

I manually set the optimizer.lr without using any scheduler and without using kvstore.set_optimizer to update the optimizer to the kvstore. Just like some codes below.
```python
optimizer = mx.optimizer.create('sgd',
                                    rescale_grad=1.0 / batch_size,
                                    learning_rate=0.0005,  # don't forget to change the learning when finetuned
                                    momentum=0.9,
                                    wd=0.0005)
mod.init_optimizer('device', optimizer)
mod.forward(batch)
mod.backward()
mod.update()
# after some epoch
optimizer.lr *= 0.1
```

When the lr changed, the loss of model decrease, I think the code for changing learning rate work. However, when I delving deep into mxnet, I find that it shouldn't work.

I have print the ``update_on_kvstore`` variable, which is True.

So I am a little confused. Hoping someone can answer me whether this way to change learning will work or not.

**Comment by eric-haibin-lin** [comment]
The optimizer object in kvstore is probably referring the same optimizer object you modified. This could work for single-machine training. See https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/module/module.py#L530-L545
It definitely doesn't work for multi-machine training. 

I'd suggest you use `lr_scheduler` instead to make sure the modification works for all scenarios.

**Comment by huyangc** [comment]
Yes, I think the kvstore is referring the same optimizer object when using a single machine. Maybe the wrapped object using in the C++ code sharing the same object address with python. (I am a newer of ctypes : )

And thanks for the suggestion. I think ``lr_scheduler`` is a better way to avoid some strange conditions as well.
